### PR TITLE
feat: enable GH actions to report reusing of deprecated reaction and metabolite identifiers

### PR DESCRIPTION
**Pull request by Hao-Chalmers**
### Main improvements in this PR:
* Adjust `sanityCheck` script to detect the cases of reusing deprecated reaction and metabolite identifiers, as discussed in #265.

**I hereby confirm that I have:**

- [X] Tested my code on my own computer for running the model
- [X] Selected `develop` as a target branch

### CODE REVIEW

File: code/test/sanityCheck.py [hunk 1]
```diff
@@ -54,5 +66,5 @@ def checkGeneAnnotation(genes):
     checkRxnAnnotation(rxns)
     checkMetAnnotation(mets)
     checkGeneAnnotation(genes)
-    print("Everything passed")
+    print("Everything is passed.")
```
Comment (mihai-sysbio) [inline code/test/sanityCheck.py:69]:
```suggestion
    print("All checks have passed.")
```

File: code/test/sanityCheck.py [hunk 2]
```diff
@@ -24,28 +35,29 @@ def checkRxnAnnotation(rxns):
     """
     check consistency of rxn lists between model and annotation file
     """
-    rxnAssoc = pd.read_table("model/reactions.tsv")
-    rxnList = rxnAssoc['rxns'].to_list()
+    rxnList = get_column_from_tsv("model/reactions.tsv", "rxns")
+    spontaneous = get_column_from_tsv("model/reactions.tsv", "spontaneous", False)
+    rxnDeprecated = get_column_from_tsv("data/deprecatedIdentifiers/deprecatedReactions.tsv", "rxns")
+    assert not bool(set(rxnList) & set(rxnDeprecated)), "Deprecated reaction reused!"
```
Comment (mihai-sysbio) [inline code/test/sanityCheck.py:41]: I think it reads easier:

```suggestion
    assert set(rxnList).isdisjoint(set(rxnDeprecated)), "Deprecated reaction reused!"
```

File: code/test/sanityCheck.py [hunk 3]
```diff
@@ -24,28 +35,29 @@ def checkRxnAnnotation(rxns):
...
 
 def checkMetAnnotation(mets):
     """
     check consistency of met lists between model and annotation file
     """
-    metAssoc = pd.read_table("model/metabolites.tsv")
-    metList = metAssoc['mets'].to_list()
+    metList = get_column_from_tsv("model/metabolites.tsv", "mets")
+    metDeprecated = get_column_from_tsv("data/deprecatedIdentifiers/deprecatedMetabolites.tsv", "mets")
+    assert not bool(set(metList) & set(metDeprecated)), "Deprecated metabolite reused!"
```
Comment (mihai-sysbio) [inline code/test/sanityCheck.py:52]: Same thinking as above:

```suggestion
    assert set(metList).isdisjoint(set(metDeprecated)), "Deprecated metabolite reused!"
```

**Review by mihai-sysbio — APPROVED** [review]
Good timing with this nice addition, I have only minor suggestions.

**Comment by Hao-Chalmers** [comment]
accepted

**Pull request merged**